src/main/java/com/group12/ciserver/database/DatabaseWrapper.java

```diff
@@ -0,0 +1,193 @@
+package com.group12.ciserver.database;
+
+import com.group12.ciserver.model.BuildInfo;
+
+import java.text.DateFormat;
+import java.util.Date;
+import java.sql.*;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+
+public class DatabaseWrapper {
+    private static Connection connection;
+    private static boolean isSetup = false;
+    public DatabaseWrapper()
+    {
+        this("ci-server.db");
+    }
+
+    public DatabaseWrapper(String dbPath) {
+        // reconnecting all the time causes the DB to hang,
+        // so do it once
+        if (!isSetup)
+        {
+            try {
+                connection = DriverManager.getConnection("jdbc:sqlite:" + dbPath);
+                createBuildTableIfNotExists();
+                isSetup = true;
+            } catch (SQLException e) {
+                System.err.println(e.getMessage());
+            }
+        }
+    }
+
+    public void createBuildTableIfNotExists() {
+        String existCheck = "SELECT count(name) FROM sqlite_master WHERE type='table' AND name='builds'";
+        String sql = """
+                CREATE TABLE builds (
+                        uid INTEGER PRIMARY KEY,
+                        commit_hash TEXT,
+                        content TEXT,
+                        date TIMESTAMP DEFAULT CURRENT_TIMESTAMP
```

Comment:
Is `TIMESTAMP` really a data type in SQLite? I have never used SQLite before, that's why I'm asking. Since I cannot find it here: https://www.sqlite.org/datatype3.html

I would recommend using `timestamp` as the column name and not `date`, since `timestamp` usually includes the data while `date` usually doesn't include the time.

One more question:
Is `DEFAULT CURRENT_TIMESTAMP` used to automatically create a timestamp or does it create the timestamp if none is provided?

Reply:
> Is `TIMESTAMP` really a data type in SQLite? I have never used SQLite before, that's why I'm asking. Since I cannot find it here: https://www.sqlite.org/datatype3.html
> 
> I would recommend using `timestamp` as the column name and not `date`, since `timestamp` usually includes the data while `date` usually doesn't include the time.
> 
> One more question: Is `DEFAULT CURRENT_TIMESTAMP` used to automatically create a timestamp or does it create the timestamp if none is provided?

Apparently sqlite doesn't have a datatype for dates/timestamps.. I guess the schema may just currently work as if it was a TEXT field. I can change the name to timestamp.

DEFAULT is only used if no value is specified, so it doesn't hurt.

However, CURRENT_TIMESTAMP in sqlite doesn't include microseconds like java Date() does.